addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/discovery/SilvercrestWifiSocketDiscoveryService.java

```diff
@@ -0,0 +1,121 @@
+ *
+ */
+public class SilvercrestWifiSocketDiscoveryService extends AbstractDiscoveryService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SilvercrestWifiSocketDiscoveryService.class);
+    private final SilvercrestWifiSocketMediator mediator;
+
+    /**
+     * Constructor of the discovery service.
+     *
+     * @throws IllegalArgumentException if the timeout < 0
+     */
+    public SilvercrestWifiSocketDiscoveryService() throws IllegalArgumentException {
+        super(SilvercrestWifiSocketBindingConstants.SUPPORTED_THING_TYPES_UIDS,
+                SilvercrestWifiSocketBindingConstants.DISCOVERY_TIMEOUT_SECONDS);
+        LOG.debug("SilvercrestWifiSocketMediator is not initialized yet will create one mediator...");
+        this.mediator = new SilvercrestWifiSocketMediator(this);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SilvercrestWifiSocketBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    protected void startScan() {
+        LOG.debug("Don't need to start new scan... background scanning in progress by mediator.");
+    }
+
+    /**
+     * Method called by mediator, when receive one packet from one unknown Wifi Socket.
+     *
+     * @param macAddress the mack address from the device.
+     * @param hostAddress the host address from the device.
+     */
+    public void discoveredWifiSocket(final String macAddress, final String hostAddress) {
+        Map<String, Object> properties = new HashMap<>(2);
+        properties.put(SilvercrestWifiSocketBindingConstants.MAC_ADDRESS_ARG, macAddress);
+        properties.put(SilvercrestWifiSocketBindingConstants.HOST_ADDRESS_ARG, hostAddress);
+
+        ThingUID newThingId = this.getNewThingId(macAddress);
+        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(newThingId).withProperties(properties)
+                .withLabel("Silvercrest Wifi Socket " + newThingId.getId()).build();
+
+        LOG.debug("Discovered new thing with mac address '{}' and host address '{}'", macAddress, hostAddress);
+
+        this.thingDiscovered(discoveryResult);
+    }
+
+    /**
+     * Gets one unrepeated {@link ThingUID} based on the existing {@link ThingUID} in
+     * {@link SilvercrestWifiSocketMediator}.
+     *
+     * @param macAddress the mac address to generate the {@link ThingUID}.
+     * @return the new unexisting {@link ThingUID}.
+     */
+    private ThingUID getNewThingId(final String macAddress) {
+        ThingUID thingUID = new ThingUID(SilvercrestWifiSocketBindingConstants.THING_TYPE_WIFI_SOCKET, macAddress);
+
+        boolean thingUIDExists = this.thingUIDExistsInMediator(thingUID);
+
+        if (thingUIDExists) {
+            int index = 1;
+            while (this.thingUIDExistsInMediator(thingUID)) {
+                thingUID = new ThingUID(SilvercrestWifiSocketBindingConstants.THING_TYPE_WIFI_SOCKET,
+                        macAddress + "_" + index);
```

Comment:
Yes, you are correct. This code actually doesn't anything because the discoveredWifiSocket method only is called if the socket doesn't exists. Will fix this.